plugins/modules/source_control/github/github_repo.py

```diff
@@ -142,16 +142,18 @@ def create_repo(gh, name, organization=None, private=False, description='', chec
     except UnknownObjectException:
         if not check_mode:
             repo = target.create_repo(
-                name=name, private=private, description=description)
+                name=name, private=private or GithubObject.NotSet, description=description or GithubObject.NotSet)
```

Comment:
Note that `or` also uses the second value for falsish values, i.e. `private == False` itself or `description == ''`. I don't think that this is intended.

I guess you should use `GithubObject.NotSet if private is None else private` and similar for `description`.